main.py

```diff
@@ -0,0 +1,58 @@
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from db.engine import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/books/", response_model=list[schemas.Book])
+def read_books(skip: int = 0, limit: int = 5, db: Session = Depends(get_db)):
+    return crud.get_all_books(skip=skip, limit=limit, db=db)
+
+
+@app.get("/books/{author_id}", response_model=list[schemas.Book])
+def filter_books(author_id: int, db: Session = Depends(get_db)):
+    return crud.get_books_by_author_id(author_id=author_id, db=db)
+
+
+@app.post("/books/create/", response_model=schemas.Book)
+def create_book(
+        book: schemas.BookCreate,
+        db: Session = Depends(get_db),
+):
+    return crud.create_book(book=book, db=db)
+
+
+@app.get("/authors/", response_model=list[schemas.Author])
+def read_author(skip: int = 0, limit: int = 5, db: Session = Depends(get_db)):
+    return crud.get_all_authors(skip=skip, limit=limit, db=db)
+
+
+@app.get("/authors/{author_id/", response_model=schemas.Author)
```

Comment:
There is a syntax error in the URL path for the `retrieve_author` endpoint. The path should be `/authors/{author_id}` instead of `/authors/{author_id/`. Make sure to close the curly brace to correctly capture the `author_id` parameter.